### PR TITLE
fix: missing supply for add-genesis-account

### PR DESCRIPTION
**Pull request by mmsqe**

### CODE REVIEW

File: module/cmd/gravity/cmd/genaccounts_test.go [hunk 1]
```diff
@@ -80,6 +85,21 @@ func TestAddGenesisAccountCmd(t *testing.T) {
...
+				var appState map[string]json.RawMessage
+				err = json.Unmarshal(genDoc.AppState, &appState)
+				require.NoError(t, err)
+				bankGenStateBz := appState[banktypes.ModuleName]
+				var bankGenState banktypes.GenesisState
+				depCdc := clientCtx.Codec
+				cdc := depCdc.(codec.Codec)
+				cdc.MustUnmarshalJSON(bankGenStateBz, &bankGenState)
+				require.Equal(t, bankGenState.Supply.String(), tc.denom)
 			}
```
Comment (cbrit) [inline module/cmd/gravity/cmd/genaccounts_test.go:103]:
```suggestion
			} else {
				require.NoError(t, cmd.ExecuteContext(ctx))

				genFile := serverCtx.Config.GenesisFile()
				bytes, err := os.ReadFile(genFile)
				require.NoError(t, err)

				var genDoc tmtypes.GenesisDoc
				tmjson.Unmarshal(bytes, &genDoc)

				var appState map[string]json.RawMessage
				err = json.Unmarshal(genDoc.AppState, &appState)
				require.NoError(t, err)

				var bankGenState banktypes.GenesisState
				bankGenStateBz := appState[banktypes.ModuleName]
				clientCtx.Codec.MustUnmarshalJSON(bankGenStateBz, &bankGenState)

				require.Equal(t, bankGenState.Supply.String(), tc.denom)
			}
```

File: module/cmd/gravity/cmd/genaccounts_test.go [hunk 2]
```diff
@@ -69,7 +74,7 @@ func TestAddGenesisAccountCmd(t *testing.T) {
 			ctx = context.WithValue(ctx, client.ClientContextKey, &clientCtx)
 			ctx = context.WithValue(ctx, server.ServerContextKey, serverCtx)
 
-			cmd := simcmd.AddGenesisAccountCmd(home)
+			cmd := AddGenesisAccountCmd(home)
```
Comment (cbrit) [inline module/cmd/gravity/cmd/genaccounts_test.go:77]: Why does this need to be changed from the `simcmd` method? The test passes without this change.

File: module/cmd/gravity/cmd/genaccounts.go [hunk 3]
```diff
@@ -151,7 +151,7 @@ contain valid denominations. Accounts may optionally be supplied with vesting pa
 			bankGenState := banktypes.GetGenesisStateFromAppState(depCdc, appState)
 			bankGenState.Balances = append(bankGenState.Balances, balances)
 			bankGenState.Balances = banktypes.SanitizeGenesisBalances(bankGenState.Balances)
-
+			bankGenState.Supply = bankGenState.Supply.Add(balances.Coins...)
```
Comment (cbrit) [inline module/cmd/gravity/cmd/genaccounts.go:154]: Could you provide the motivation for setting this? Is supply not calculated automatically from the balances?

**Review by cbrit — CHANGES_REQUESTED** [review]
Thanks for the PR! A few comments and question!

File: module/cmd/gravity/cmd/genaccounts_test.go (hunk 2 above)
Comment (mmsqe) [reply module/cmd/gravity/cmd/genaccounts_test.go:76]: It was testing `AddGenesisAccountCmd` from sdk, which already covered by its own [test](https://github.com/cosmos/cosmos-sdk/blob/3a7a2645f05f0aecdf96ebda14a7c62f6d4ddc55/x/genutil/client/cli/genaccount_test.go#L25)

File: module/cmd/gravity/cmd/genaccounts.go (hunk 3 above)
Comment (mmsqe) [reply module/cmd/gravity/cmd/genaccounts.go:154]: I find this [unalignment](https://github.com/cosmos/cosmos-sdk/blob/3a7a2645f05f0aecdf96ebda14a7c62f6d4ddc55/x/auth/helpers/genaccounts.go#L122) when fix `panic: genesis supply is incorrect` when fix [sim](https://github.com/crypto-org-chain/gravity-bridge/pull/218) tests

**Review by cbrit — APPROVED** [review]
LGTM

**Pull request merged**